### ISSUE TITLE
SIMD: Batch weight and bias gradient accumulation

### ISSUE THREAD

**Issue by stservice** [stSoftwareAU/NEAT-TS#1214]
## Background

During backpropagation, weight and bias gradients are accumulated across the mini-batch. The current implementation in TypeScript processes each synapse update individually, which could benefit from SIMD batching.

## Current State

\`src/propagate/Weight.ts\`:
```typescript
export function accumulateWeight(
  synapse: Synapse,
  fromActivation: number,
  toValue: number,
  averageWeight: number,
  options: BackPropagationOptions
): number {
  // Single synapse weight adjustment
  const delta = targetWeight - synapse.weight;
  // Track statistics: totalPositiveActivation, totalNegativeActivation, etc.
  synapse.adjustedCount++;
  return adjust;
}
```

\`src/propagate/Bias.ts\`:
```typescript
export function adjustedBias(
  neuron: Neuron,
  error: number,
  // ...
): number {
  // Single neuron bias adjustment
}
```

## Proposal

Implement batched gradient accumulation that processes 4 synapses or neurons simultaneously.

### Batch Weight Accumulation (WASM)
```rust
pub fn accumulate_weights_batch_4way(
    synapse_indices: &[u32; 4],
    from_activations: &[f32; 4],
    to_values: &[f32; 4],
    average_weights: &[f32; 4],
    // ... other parameters
) -> [f32; 4] {
    // Parallel weight adjustment for 4 synapses
    let deltas = f32x4_sub(targets, weights);
    // Parallel clipping, scaling, accumulation
}
```

### Batch Bias Adjustment (WASM)
```rust
pub fn adjust_biases_batch_4way(
    neuron_indices: &[u32; 4],
    errors: &[f32; 4],
    learning_rate: f32,
    // ... other parameters
) -> [f32; 4] {
    // Parallel bias adjustment for 4 neurons
}
```

## Expected Benefits

- **4x speedup** for weight gradient accumulation phase
- Statistics tracking (sum, count) can be done in SIMD
- Training loops already expect mini-batch processing (default \`batchSize: 64\`)
- Independent synapse adjustments parallelise perfectly

## Implementation Notes

1. Could be implemented in WASM for better performance
2. Alternatively, TypeScript with typed arrays could benefit from V8 SIMD optimisations
3. Min/max operations for weight limiting vectorise well
4. Statistics counters may need atomic operations if multi-threaded

## Files to Modify

- \`wasm_activation/src/lib.rs\`: Add batch accumulation functions (if WASM approach)
- \`src/propagate/Weight.ts\`: Batch variant (if TypeScript approach)
- \`src/propagate/Bias.ts\`: Batch variant (if TypeScript approach)

## Success Criteria

- 4x speedup for gradient accumulation in benchmarks
- All training tests pass
- Weight/bias convergence matches single-synapse implementation
- Statistics tracking remains accurate

Related to: #1205, #1212, #1213
